### PR TITLE
Updated Deep Learning (DL) workflows to use the most recent DL tools version

### PR DESCRIPTION
**Pull request by kxk302**

### CODE REVIEW

**Review by hexylena — COMMENTED** [review]
It looks like you're using the tutorial mode tool linking in the tutorials, since it looks like you're changing at least one tool id, can you please check the associated tutorials to make sure they're using the correct IDs as well?

This was the one I noticed, there may be more.
"toolshed.g2.bx.psu.edu/repos/bgruening/keras_model_config/keras_model_config/0.5.0",

**Comment by kxk302** [comment]
> It looks like you're using the tutorial mode tool linking in the tutorials, since it looks like you're changing at least one tool id, can you please check the associated tutorials to make sure they're using the correct IDs as well?
> 
> This was the one I noticed, there may be more. "toolshed.g2.bx.psu.edu/repos/bgruening/keras_model_config/keras_model_config/0.5.0",

@hexylena Good catch! I updated the tool versions in tutorials and pushed the change.

**Comment by anuprulez** [comment]
I will test these workflows as well

**Comment by anuprulez** [comment]
I tried to run these workflows on the EU but the jobs have not started so far. I guess they are in queue waiting for GPUs. I will try on Org

**Comment by anuprulez** [comment]
Summary of my review:

I tried to run these workflows on both servers, the EU and Org.

- On the EU, the jobs stayed in the queue and did not start after data upload. I think it is because of its necessity to run on GPUs and possibly there are no free ones at the moment. It can be fixed by moving from GPUs to CPUs. At the moment, we don't have that many GPUs (on EU). 

- On Org:

1. the FNN workflow worked well and the results are comparable to the ones mentioned in its respective tutorial (https://training.galaxyproject.org/training-material/topics/statistics/tutorials/FNN/tutorial.html). Its history on Org: https://usegalaxy.org/u/anupkumar/h/fnn-test

2. the CNN workflow also finishes but the final confusion plot shows classification numbers that are significantly different from the ones mentioned in the tutorial (https://training.galaxyproject.org/training-material/topics/statistics/tutorials/CNN/tutorial.html#classification-of-mnist-dataset-images-with-cnn). its history: https://usegalaxy.org/u/anupkumar/h/cnn-test. Could be because of the smaller test set compared to what was used to generate the plot mentioned in the tutorial? For example, for Class 1, the tutorial has over 1000 correctly predicted samples but the plot in the above history has just over 650. It exists for other classes as well.  

3. While running the RNN workflow, I am not able to upload one of the datasets (`X_train`) because I think it is big and Org has returned the following error twice: `Failed to fetch URL https://zenodo.org/record/4477881/files/X_train.tsv. The read operation timed out`. I could not find the associated datasets in the data library. Its history: https://usegalaxy.org/u/anupkumar/h/rnn-test. I am not sure how this could be fixed. Did not have such issue with the EU.

ping @kxk302

**Review by anuprulez — COMMENTED** [review]
Summary of my review:

I tried to run these workflows on both servers, the EU and Org.

- On the EU, the jobs stayed in the queue and did not start after data upload. I think it is because of its necessity to run on GPUs and possibly there are no free ones at the moment. It can be fixed by moving from GPUs to CPUs. At the moment, we don't have that many GPUs (on EU). 

- On Org:

1. the FNN workflow worked well and the results are comparable to the ones mentioned in its respective tutorial (https://training.galaxyproject.org/training-material/topics/statistics/tutorials/FNN/tutorial.html). Its history on Org: https://usegalaxy.org/u/anupkumar/h/fnn-test

2. the CNN workflow also finishes but the final confusion plot shows classification numbers that are significantly different from the ones mentioned in the tutorial (https://training.galaxyproject.org/training-material/topics/statistics/tutorials/CNN/tutorial.html#classification-of-mnist-dataset-images-with-cnn). its history: https://usegalaxy.org/u/anupkumar/h/cnn-test. Could be because of the smaller test set compared to what was used to generate the plot mentioned in the tutorial? For example, for Class 1, the tutorial has over 1000 correctly predicted samples but the plot in the above history has just over 650. It exists for other classes as well.  

3. While running the RNN workflow, I am not able to upload one of the datasets (`X_train`) because I think it is big and Org has returned the following error twice: `Failed to fetch URL https://zenodo.org/record/4477881/files/X_train.tsv. The read operation timed out`. I could not find the associated datasets in the data library. Its history: https://usegalaxy.org/u/anupkumar/h/rnn-test. I am not sure how this could be fixed. Did not have such issue with the EU.

ping @kxk302

**Comment by kxk302** [comment]
@anuprulez I tried uploading the files for RNN tutorial. I got an error uploading it, but re-tried and it succeeded. I then ran the workflow and got similar results as in the tutorial. So, I think we are good for FNN and RNN workflows/tutorials.

As for the CNN tutorial, I guess removing the 'padding="same"' maybe the culprit. I am investigating that problem right now. Will let you know when I resolve it.

**Comment by anuprulez** [comment]
@kxk302 Is the issue related to `padding="same"` for CNN tutorial fixed by this PR: https://github.com/bgruening/galaxytools/pull/1201? Unfortunately, a few tests are still failing in this PR.

**Comment by kxk302** [comment]
> @kxk302 Is the issue related to `padding="same"` for CNN tutorial fixed by this PR: [bgruening/galaxytools#1201](https://github.com/bgruening/galaxytools/pull/1201)? Unfortunately, a few tests are still failing in this PR. I think this was the only problem left to be fixed (CNN workflow/tutorial), the other two workflows worked fine.

@anuprulez yes, it's a fix for padding='same' issue. I see all checks as being green. Can you please post a link to the failing test? Thanks!

**Comment by anuprulez** [comment]
I think once these test passes (from the PR: https://github.com/bgruening/galaxytools/pull/1201) and updated tools are installed, the PR (https://github.com/galaxyproject/training-material/pull/3292) will automatically start working. I will verify this once the new versions of tools (from the PR: https://github.com/bgruening/galaxytools/pull/1201) are installed. Thanks for working on this @kxk302

**Review by shiltemann — APPROVED** [review]

**Comment by shiltemann** [comment]
Hi @kxk302 and @anuprulez, can this one be merged now? :)

**Comment by kxk302** [comment]
> Hi @kxk302 and @anuprulez, can this one be merged now? :)

Yes, please!
